### PR TITLE
Milestone: Members-Only

### PR DESCRIPTION
**Pull request by enkog**
This project brings to light the practice of performing authentication and authorization using rails own gem called devise. 
I have employed the use of rails MVC structure to achieve this.

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] Couldnt run the app because of the error below.Check your migration file
````ruby
Caused by:
NameError: undefined local variable or method `t' for #<AddUserIdToPosts:0x000055ff5956f4d8>

````
# \[OPTIONAL\] 

* [ ] A helper method to check if user is signed_in? can be handy here

```ruby
# must be a helper method to be used in views
helper_method :signed_in?
def signed_in?
  !current_user.nil?
end
```
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @enkog   ! :wave: :wave:

#### You have done well so far :clap: However, there are changes required for you to make in order for this project to be approved. Kindly work on them and submit them for another code review.


**Recommended**

- [x] Just like the previous TSE, I, unfortunately, could not run your project due to the error below, please check your migration file.

**Caused by:
NoMethodError: undefined method `add_references' for #<AddUserIdToPosts:0x00007fffd5d90c88>**


__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :posts, only: %i[new create index]
 end
```
Comment (peterrobert) [inline config/routes.rb]: ### [OPTIONAL]
This is a very good practice: Kindly add root route. When a user visits your application it is good to have a root route other than being redirected to the default " your on rails"  template

Use this [article](https://curriculum.railsbridge.org/intro-to-rails/setting_the_default_page) to learn more about this.

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED ♻️ ⏹️ 

Hello, @enkog nice progress you are almost there 👍🏿 . However, there are a few changes that need to be made before we can continue. Kindly check below ⬇️ 

- [x] There should be sign in and sign out links present in your application. At the moment I had to directly type the weblinks in the browser. Kindly add some navigation. 
sign_in_out_link_requirement

- [x] A user should be able to sign out. At the moment there is no way for a user to sign out. This can all be solved if you include links to your application.

sign_out_requirement

- [x] As a guest user, you can only see post's text. Kindly fix this too. I can see that you have restricted this in your controller 👍🏿 , however, in your views I have no way of knowing this. This will all be solved if you have some navigation in your application.

guest_can_see_only_posts_requirement

- [ ] In your application helper, you need to define the below method ⬇️ 

`def signed_in_only!
  redirect_to  new_user_session_path unless current_user
end
`
This will check if it is the current_user.

### if you have any question kindly reach me via slack @peterrobert

### HAPPY CODING 💻

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****


 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**